request-manager/graph/grapher.go

```diff
@@ -177,25 +189,32 @@ func getMissingSets(actual map[string]bool, declared []*spec.NodeSet) []string {
 	return missing
 }
 
-// Builds a sequence graph and performs checks on that sequence.
+// buildSeqGraph builds a sequence graph and performs checks on that sequence.
 // It does not analyze any subsequences. As such, it can't e.g. check validity
 // of `sets` declarations in sequence and conditional nodes.
-// Returns the graph, the set of job args its component nodes set, and an error if any should occur.
+// Returns the graph, the set of job args its component nodes set, and an error
+// if any should occur.
 func buildSeqGraph(seqSpec *spec.Sequence, idgen id.Generator) (seqGraph *Graph, sets map[string]bool, err error) {
 	// The graph we'll be filling in
-	seqGraph, err = newGraph(seqSpec.Name, idgen)
+	seqGraph, err = newSeqGraph(seqSpec.Name, idgen)
 	if err != nil {
 		return nil, nil, err
 	}
-	// newGraph doesn't manager `Order` for us; we'll have to maintain it ourselves in this function
+
+	// We have to maintain `Order` ourselves
 	seqGraph.Order = append(seqGraph.Order, seqGraph.Source)
 
-	// Set of job args available, i.e. sequence args + args set by nodes in graph so far
+	// Set of job args available, i.e. sequence args + args set by nodes in
+	// graph so far.
+	// We start out with just the sequence args.
 	jobArgs := getAllSequenceArgs(seqSpec)
 
-	// Create a graph with a single node for every node in the spec
-	// Key on node names--they should be unique within a sequence (otherwise, dependencies
-	// are ill-defined)
+	// Create a graph consisting of a single node for every node in the spec.
+	// There's a function to insert a subgraph into a graph, but not to insert
+	// a node into a graph. Creating these as single-node graphs upfront rather
+	// than nodes makes the code more readable later.
+	// Key on node names; they should be unique within a sequence (otherwise,
+	// dependencies are ill-defined).
```

Comment:
Pretty good comment, but after a few close reads and reading the actual code block, I think it'd be more clear as:
```
	// Create graph nodes for every node in the sequence spec. We're not connecting
	// the nodes yet (i.e. not building a graph), just initializing all the graph
	// nodes. In the next loop, we'll wire them up (i.e. build the sequence graph).
	//
	// Note: graph nodes can be any type (job, seq, or conditional), but for
	// sequence graphs we don't do anything (i.e. don't create job.Job for job node);
	// a Resolver does that when creating a request graph (RG). Sequence graphs
	// are only (graph) structure and metadata about that structure.
```